pkg/server/auth/server_test.go

```diff
@@ -625,6 +633,69 @@ func TestUserInfoOIDCFlow(t *testing.T) {
 	g.Expect(info.Email).To(Equal("jane.doe@example.com"))
 }
 
+func TestRefresh(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	tokenSignerVerifier, err := auth.NewHMACTokenSignerVerifier(5 * time.Minute)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	s, m := makeAuthServer(t, nil, tokenSignerVerifier, []auth.AuthMethod{auth.OIDC})
+
+	tokens := getVerifyTokens(t, s, m)
+
+	tf := tokens["refresh_token"].(string)
+
+	req := httptest.NewRequest(http.MethodGet, "https://example.com/test", nil)
+	req.AddCookie(&http.Cookie{
+		Name:  auth.RefreshTokenCookieName,
+		Value: tf,
+	})
+
+	w := httptest.NewRecorder()
+	cookies := make(map[string]*http.Cookie)
+
+	g.Expect(s.Refresh(w, req)).To(Succeed())
+
+	for _, c := range w.Result().Cookies() {
+		if c.Name == auth.IDTokenCookieName || c.Name == auth.AccessTokenCookieName || c.Name == auth.RefreshTokenCookieName {
+			cookies[c.Name] = c
+		}
+	}
+
+	g.Expect(cookies).To(HaveKey(auth.IDTokenCookieName))
+	g.Expect(cookies).To(HaveKey(auth.AccessTokenCookieName))
+	g.Expect(cookies).To(HaveKey(auth.RefreshTokenCookieName))
+
+	_, err = m.Keypair.VerifyJWT(cookies[auth.IDTokenCookieName].Value)
```

Comment:
These tests are a bit complex to understand, can we either split them, or at least comment to explain why the three _almost_ identical calls are made?